docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,555 @@
+}
+```
+
+</details>
+
+### Step 3: Deploy the Extension Contract
+
+We will use the abi and bytecode to deploy the extension contract
+
+```js
+import { abi as TipMeABI } from './TipMe.json';
+import { bytecode as TipMeBytecode } from './TipMe.json';
+
+const TipMeFactory = new ethers.ContractFactory(
+  TipMeABI,
+  TipMeBytecode,
+  signer,
+);
+
+const tipMeContract = await TipMeFactory.deploy();
+await tipMeContract.deployTransaction.wait();
+
+const tipMeExtensionAddress = tipMeContract.addres;
+```
+
+### Step 4: Encode Function Selector and Store in Universal Profile
+
+:::info Requirements
+
+The address calling the `setData(..)` function needs to have `ADDEXTENSIONS` and `CHANGEEXTENSIONS` permission otherwise the call will fail. Check the [keyManager guides](../key-manager/get-controller-permissions.md) to learn more about permissions.
+
+:::
+
+We need to encode the function selector of `tipMe(..)` and store it in the Universal Profile with a specific data key according to the [LSP17-ContractExtension](../../../standards/generic-standards/lsp17-contract-extension.md) standard.
+
+#### Prepare data key and data value
+
+```js
+import { ERC725YDataKeys } from '@lukso/lsp-smart-contracts';
+
+// We get `TipMeABI` from the last step
+// import {abi as TipMeABI} from './TipMe.json';
+const tipMeInterface = new ethers.utils.Interface(TipMeABI);
+const tipMeFunctionSelector = tipMeInterface.getSighash('tipMe()');
+
+// Define the DataKey for the extension
+// according to the LSP17-ContractExtension standard
+const extensionDataKey =
+  ERC725YDataKeys.LSP17.LSP17ExtensionPrefix +
```

Comment:
Like suggested to @skimaharvey for his LSP23 guide, I would suggest here to use _erc725.js_ to encode the data key. It will be easier to do rather than having to do it manually.

Also agree with @Hugoo on putting a short link on how the data key is generated.

Reply:
I'll add a comment, maybe not worth having one more dependency just for the sake of this key